### ISSUE TITLE
Variants on Text cause issue with ESLint and cannot build project

### ISSUE THREAD

**Issue by flavienbonvin** [chakra-ui/chakra-ui#5370]
### Description

When creating a custom theme for the Text component, I except to be able to have variants but I have ESLint error informing me that this is not possible

### Link to Reproduction

https://codesandbox.io/s/nervous-alex-uow1k

### Steps to reproduce

1. Go to provided codesandbox demo
2. Once the project is started go on `index.tsx` component
3. The variant is correctly applied to the `Text`

### Chakra UI Version

1.7.4

### Browser

Brave 97

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

Variants on the `Text` components causing an error when trying to build the project or deploying it on Vercel. The variants are correctly displayed on the screens (as showed in the codesandbox) but it's not possible to build the project (using `next build`).

The error is the following: `Type error: Type 'string' is not assignable to type 'undefined'.``

It's possible to fix the issue by using textStyles instead but this is not consistent with other components that all use variants (even the `Heading` component support variants).

**Comment by niels-k-86** [comment]
The same goes for the `size` attribute on `<Text>`

**Comment by primos63** [comment]
The issue lies in system.types.d.ts and the interface ThemingProps. A recent change has caused the issue. 
As a temporary fix, please revert back to @chakra-ui/react version 1.7.2.

**Comment by jonathanprozzi** [comment]
Thank you for the detail on what's causing the issue and for the workaround. Very much appreciated!

**Comment by rainmak33r** [comment]
> The issue lies in system.types.d.ts and the interface ThemingProps. A recent change has caused the issue. As a temporary fix, please revert back to @chakra-ui/react version 1.7.2.

Doesn't seem to work for me. Tried several different versions.

**Comment by primos63** [comment]
> > The issue lies in system.types.d.ts and the interface ThemingProps. A recent change has caused the issue. As a temporary fix, please revert back to @chakra-ui/react version 1.7.2.
> 
> Doesn't seem to work for me. Tried several different versions. 

Did you change `@chakra-ui/react` to `"1.7.2"` in your package.json?

**Comment by rainmak33r** [comment]
yep, I'm using node 16, might that cause the problem ?

**Comment by primos63** [comment]
Don't think so. The issue was in our TypeScript code. Can you post the dependency section of your package.json?

**Comment by rainmak33r** [comment]
> Don't think so. The issue was in our TypeScript code. Can you post the dependency section of your package.json?

```
"dependencies": {
    "@chakra-ui/icons": "^1.0.16",
    "@chakra-ui/react": "1.7.2",
    "@emotion/react": "11.5.0",
    "@emotion/styled": "11.3.0",
    "@types/nodemailer": "^6.4.4",
    "aws-sdk": "2.1043.0",
    "axios": "0.24.0",
    "bcrypt": "^5.0.1",
    "body-parser": "^1.19.0",
    "chakra-react-select": "1.3.4",
    "core-js": "^3.6.5",
    "cors": "2.8.5",
    "express": "^4.17.1",
    "formik": "^2.2.9",
    "framer-motion": "^4.1.17",
    "handlebars": "^4.7.7",
    "helmet": "4.6.0",
    "jsonwebtoken": "^8.5.1",
    "lodash": "4.17.21",
    "luxon": "2.2.0",
    "multer": "1.4.3",
    "multer-s3": "2.10.0",
    "mysql": "^2.18.1",
    "mysql2": "2.3.3",
    "nodemailer": "^6.7.1",
    "nodemailer-express-handlebars": "^4.0.0",
    "promise-mysql": "5.0.4",
    "queue-promise": "2.2.1",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-query": "3.34.2",
    "react-router-dom": "5.3.0",
    "react-select": "5.2.1",
    "regenerator-runtime": "0.13.7",
    "sequelize": "6.12.0-beta.3",
    "tslib": "^2.0.0",
    "uuid": "^8.3.2",
    "web-vitals": "^2.1.2",
    "yup": "^0.32.11"
  },
  ```
  
  and `npm list @chakra-ui/react`
![image](https://user-images.githubusercontent.com/14319852/149560371-c0d0991c-ec30-462c-8293-d21a58c8dc89.png)

**Comment by rainmak33r** [comment]
Only after I manually installed `@chakra-ui/system@1.8.2` the issue was resolved. But when installing a clean `@chakra-ui/react@1.7.2` it installed the wrong version.

**Comment by primos63** [comment]
That's interesting.  Not sure why you were getting that. Did you also try removing `node_modules` and running `npm install`? I ran that and did not have the issue with 1.7.2 or 1.7.3.

I tried the CodeSandbox again and the error goes away with either v1.7.3 or earlier. I first said it broke in 1.7.3, but that doesn't appear to be the case. Seems you have to wait a few seconds after changing versions for TypeScript to kick in. I may not have done that for 1.7.3.
I went over to StackBlitz and created a Next app with 1.7.2 and 1.7.3 and it worked.

**Comment by rainmak33r** [comment]
I believe the problem was in `chakra-react-select` package, it has `"@chakra-ui/react": "^1.6.0",` in dependencies, could it install the right version of `@chakra-ui/react` and just grab everything else of the latest versions ?

**Comment by yotam707** [comment]
This error also happens for **every**  size property (in my use case it's also happens in Button)

**Comment by masashi-sutou** [comment]
> System v1.10.0
> - Updated type ThemingProps to allow string values for the props variant and size even on components which are not in the default theme.
> 
> https://github.com/chakra-ui/chakra-ui/blob/main/CHANGELOG.md#25-01-2022

- I think this issue was solved by @ chakra-ui / react` v1.8.0`.

**Comment by TimKolberger** [comment]
Thanks @masashi-sutou, this was fixed in 1.8.0 👍
